src/category_theory/bicategory/natural_transformation.lean

```diff
@@ -0,0 +1,256 @@
+    (F.map_comp f g ▷ _) ≫ (α_ _ _ _).hom ≫ (_ ◁ naturality g) ≫ (α_ _ _ _).inv ≫
+      (naturality f ▷ G.map g) ≫ (α_ _ _ _).hom . obviously)
+
+restate_axiom oplax_nat_trans.naturality_naturality'
+restate_axiom oplax_nat_trans.naturality_comp'
+restate_axiom oplax_nat_trans.naturality_id'
+attribute [simp, reassoc] oplax_nat_trans.naturality_naturality
+  oplax_nat_trans.naturality_comp oplax_nat_trans.naturality_id
+
+namespace oplax_nat_trans
+
+section
+variables (F : oplax_functor B C)
+
+/-- The identity oplax natural transformation. -/
+@[simps]
+def id : oplax_nat_trans F F :=
+{ app := λ a, 𝟙 (F.obj a),
+  naturality := λ a b f, (ρ_ _).hom ≫ (λ_ _).inv,
+  naturality_naturality' := λ a b f f' η, by
+  { rw [assoc, ←left_unitor_inv_naturality, ←right_unitor_naturality_assoc] },
+  naturality_comp' := λ a b c f g, by
+  { rw [assoc, ←left_unitor_inv_naturality, ←right_unitor_naturality_assoc],
+    simp only [triangle_assoc_comp_right_assoc, right_unitor_comp, left_unitor_comp_inv,
+      whisker_right_comp, inv_hom_whisker_left_assoc, assoc, whisker_left_comp] },
+  naturality_id' := λ a, by
+  { rw [assoc, ←left_unitor_inv_naturality, ←right_unitor_naturality_assoc,
+      unitors_equal, unitors_inv_equal] } }
+
+instance : inhabited (oplax_nat_trans F F) := ⟨id F⟩
+
+variables {F} {G H : oplax_functor B C} (η : oplax_nat_trans F G) (θ : oplax_nat_trans G H)
+
+section
+variables {a b c : B} {a' : C}
+
+@[simp, reassoc]
+lemma whisker_left_naturality_naturality (f : a' ⟶ G.obj a) {g h : a ⟶ b} (β : g ⟶ h) :
+  (f ◁ (G.map₂ β ▷ θ.app b)) ≫ (f ◁ θ.naturality h) =
+    (f ◁ θ.naturality g) ≫ (f ◁ (θ.app a ◁ H.map₂ β)) :=
+by simp only [←whisker_left_comp, naturality_naturality]
+
+@[simp, reassoc]
+lemma whisker_right_naturality_naturality {f g : a ⟶ b} (β : f ⟶ g) (h : G.obj b ⟶ a') :
+  ((F.map₂ β ▷ η.app b) ▷ h) ≫ (η.naturality g ▷ h) =
+    (η.naturality f ▷ h) ≫ ((η.app a ◁ G.map₂ β) ▷ h) :=
+by simp only [←whisker_right_comp, naturality_naturality]
+
+@[simp, reassoc]
+lemma whisker_left_naturality_comp (f : a' ⟶ G.obj a) (g : a ⟶ b) (h : b ⟶ c) :
+  (f ◁ θ.naturality (g ≫ h)) ≫ (f ◁ (_ ◁ H.map_comp g h)) =
+    (f ◁ (G.map_comp g h ▷ _)) ≫(f ◁ (α_ _ _ _).hom) ≫
+      (f ◁ (_ ◁ θ.naturality h)) ≫ (f ◁ (α_ _ _ _).inv) ≫
+        (f ◁ (θ.naturality g ▷ H.map h)) ≫ (f ◁ (α_ _ _ _).hom) :=
+by simp only [←whisker_left_comp, naturality_comp]
+
+@[simp, reassoc]
+lemma whisker_right_naturality_comp (f : a ⟶ b) (g : b ⟶ c) (h : G.obj c ⟶ a') :
+  (η.naturality (f ≫ g) ▷ h) ≫ ((_ ◁ G.map_comp f g) ▷ h) =
+    ((F.map_comp f g ▷ _) ▷ h) ≫ ((α_ _ _ _).hom ▷ h) ≫
+      ((_ ◁ η.naturality g) ▷ h) ≫ ((α_ _ _ _).inv ▷ h) ≫
+        ((η.naturality f ▷ G.map g) ▷ h) ≫ ((α_ _ _ _).hom ▷ h) :=
+by simp only [←whisker_right_comp, naturality_comp]
+
+@[simp, reassoc]
+lemma whisker_left_naturality_id (f : a' ⟶ G.obj a) :
+  (f ◁ θ.naturality (𝟙 a)) ≫ (f ◁ (_ ◁ H.map_id a)) =
+    (f ◁ G.map_id a ▷ _) ≫ (f ◁ (λ_ _).hom) ≫ (f ◁ (ρ_ _).inv) :=
```

Comment:
Slight inconsistency: In lines 101 and 110 you use parentheses for double-sided whiskering, but here you don't. I thought it doesn't matter since they commute, but in fact you can remove parentheses on line 101 but not on 108, 110 ...

I'm also a bit surprised to find that ◁ and ▷ are right associative as opposed to left associative, so you can remove parentheses on line 102 but not on 110,111.